### ISSUE TITLE
Grid with setHeightByRows(N) is rendered wrong inside GridLayout

### ISSUE THREAD

**Issue by pleku** [vaadin/framework#8207]
Tested with v8 & Valo, when Grid using height-by-rows, it is rendered on top of the next GridLayout row.
Simplified example, originally present in _uitest/../VaadinIcons.java_
 ```java
Grid<String> grid = new Grid<>();
grid.setItems("item 1", "item 2", "item 3");
grid.addColumn(string -> string).setCaption("column 1");
grid.addColumn(string -> "Another " + string).setCaption("column 2");
grid.setHeightByRows(3);

GridLayout gl = new GridLayout(1, 3);
gl.addComponent(grid);
gl.addComponent(new Label("foobar");
```

Works as expected when removing `grid.setHeightByRows(3)`.

**Comment by haiwan** [comment]
similar problem in 7.6.6 as well
<img width="315" alt="screen shot 2017-02-24 at 16 32 31" src="https://cloud.githubusercontent.com/assets/3388675/23307109/fdea9860-faae-11e6-94db-7af996bab1af.png">

**Comment by hesara** [comment]
Most likely this will be fixed by #10077 - at least the tests that were marked as broken because of this were fixed in that branch.
